lib/matplotlib/widgets.py

```diff
@@ -2052,16 +2052,34 @@ def _on_key_release(self, event):
         """Key release event handler."""
 
     def set_visible(self, visible):
-        """Set the visibility of our artists."""
-        self.visible = visible
+        """Set the visibility of the selector artists."""
+        self._visible = visible
         for artist in self.artists:
             artist.set_visible(visible)
 
+    def get_visible(self):
+        """Get the visibility of the selector artists."""
+        return self._visible
+
+    @property
+    def visible(self):
+        return self.get_visible()
+
+    @visible.setter
+    def visible(self, visible):
+        _api.warn_deprecated(
+            "3.6", message="The visible setter was deprecated since %(since)s"
+            " and will be removed in %(removal)s. Use set_visible instead.")
```

Comment:
No need to write the whole message; just set `alternative="set_visible"`.

Reply:
Thanks, done!